unyt/exceptions.py

```diff
@@ -7,13 +7,31 @@
 
 
 class UnytError(Exception):
-    # a generic exception type that just signals "this is coming from unyt"
-    # TBD: should this be named `UnitError` instead ?
-    # TBD: should all other custom excpetion inherit from it ?
+    """
+    A generic exception type that all unyt exceptions are derived from.
+    It may be raised directly in rare situations that do not fit any existing
+    exceptions and don't justify the creation of a new one.
+
+    Since it serves as a base class for all unyt exceptions, it may be used
+    as a catch-all exception type in dependent code. For instance
+
+    >>> from unyt import cm, s
+    >>> from unyt.exceptions import UnytError
+    >>> a = 1 * cm
+    >>> b = 1 / s
+    >>> try:
+    ...     a + b
+    ... except UnytError:
+    ...     pass
+
+    However, it is generally recommended to only capture exceptions as
+    specific as possible.
+    """
+
     pass
 
 
-class UnitOperationError(ValueError):
+class UnitOperationError(UnytError, ValueError):
```

Comment:
disclaimer: using multiple inheritance is explicitly recommended against in the documentation https://docs.python.org/3/library/exceptions.html#inheriting-from-built-in-exceptions

So I'm kinda hoping that the fact `UnytError` doesn't hold any specialisation is a safe case.

Reply:
I wish that comment in the python docs had more detail about how this might fail. Let's see if we get any complaints from users after doing a release.

Reply:
agreed